catalog/files/impl/src/main/java/org/projectnessie/catalog/files/s3/S3ObjectIO.java

```diff
@@ -45,6 +45,16 @@ public S3ObjectIO(S3ClientSupplier s3clientSupplier, Clock clock) {
     this.clock = clock;
   }
 
+  @Override
+  public void ping(StorageUri uri) throws IOException {
+    S3Client s3client = s3clientSupplier.getClient(uri);
+    try {
+      s3client.getBucketLocation(b -> b.bucket(uri.requiredAuthority()));
```

Comment:
This is fine from my POV, but it probably requires extra permissions in AWS.

Reply:
For the bucket-location? Uh

Reply:
Also `S3Client` javadoc has: `This operation is not supported by directory buckets.`

Reply:
Oh! there's a `headBucket` anyways

Reply:
rebased + updated this one

Reply:
+1 to `headBucket`. MinIO does not appear to support `GetBucketLocaltion`: https://min.io/docs/minio/linux/reference/s3-api-compatibility.html